gorm-rest/src/main/groovy/yakworks/rest/gorm/controller/EntityResponder.groovy

```diff
@@ -123,12 +123,15 @@ class EntityResponder<D> {
             // when paging we need a sort so rows dont show up next page, so use id if nothing is specified.
             // see https://github.com/9ci/domain9/issues/2280
 
-            // Default sort by id:asc if no  sort is provided in params
-            //if sort was passed in params, or if $sort is provided in `q` criteria, then the default sort by id wont be used
-            //XXX @SUD hot fix, we need a test, this is what I htink is blowing up projections
-            // if(!qargs.sort) {
-            //     qargs.sort("id":"asc")
-            // }
+            /*
```

Comment:
can we make this a method in qargs so its reusable, mentioned it in one of the earlier reviews. then can just call `qargs.applySortDefault()` or something like that and it will do the basic if check logic and javadocs can have the good notes you have here. then it can be tested and be clearer along with the rest of the Qargs tests. 

better yet, this entire chunk could be simplified into builder methods instead

<img width="1057" alt="image" src="https://github.com/yakworks/gorm-tools/assets/187726/49f00219-bd02-4e1f-8eed-93e68ab96b27">

like this

```groovy
QueryArgs qargs = QueryArgs
    .of(pager)
    .qRequired(pathItem?.qRequired)
    .sortById(true)
    .build(parms)
    .validateQ()
...
```

Reply:
Fixed